### PR TITLE
Deprecate WhitelistingTextInputFormatter and BlacklistingTextInputFormatter

### PR DESCRIPTION
**Pull request by Hixie**
They are replaced by the single class FilteringTextInputFormatter.

Also, Remove deprecated_member_use_from_same_package. Since I ignore this lint rather than adding a bunch of //ignores, I also go and clean up other places we ignored this lint.

fyi @Piinks for the deprecation stuff
fyi @xster since you added these classes

### CODE REVIEW

**Comment by Hixie** [comment]
Might be easier to review by reviewing the two commits separately.

**Review by Piinks — COMMENTED** [review]
This is awesome! The analyzer has some thoughts though.

**Comment by Hixie** [comment]
Update: PR should have all the tests fixed now, and I spoke to the Gallery folks, they'll update after this lands (I made the specific deprecations that were failing on the Gallery not be failures for now; I'll deprecate them later once there's no breaking changes to worry about).

**Comment by Hixie** [comment]
@Piinks PTAL

**Comment by Hixie** [comment]
Fixes https://github.com/flutter/flutter/issues/59099

**Comment by Hixie** [comment]
Merge conflicts fixed. Should be ready to land assuming no problems are found in review.

File: packages/flutter/lib/src/services/text_formatter.dart [hunk 1]
```diff
@@ -78,34 +76,140 @@ class _SimpleTextInputFormatter extends TextInputFormatter {
...
+  /// Additionally, each segment of the string before, during, and
+  /// after the current selection in the [TextEditingValue] is handled
+  /// separately. This means that, in the case of the "Into the Woods"
+  /// example above, if the selection ended between the two "o"s in
+  /// "Woods", even if the pattern was `RegExp('o+')`, the result
+  /// would be "Int* the W**ds", since the two "o"s would be handled
+  /// in separate passes.
+  ///
+  /// See also [String.splitMapJoin], which is used to implement this
+  /// behavior in both cases.
```
Comment (goderbauer) [inline packages/flutter/lib/src/services/text_formatter.dart:212]: Nice comprehensive doc comment!

File: packages/flutter/lib/src/services/text_formatter.dart [hunk 2]
```diff
@@ -116,16 +220,87 @@ class BlacklistingTextInputFormatter extends TextInputFormatter {
...
+  @Deprecated(
+    'Use FilteringTextInputFormatter.singleLineFormatter instead. '
+    'This feature was deprecated after v1.20.0-1.0.pre.'
+  )
   static final BlacklistingTextInputFormatter singleLineFormatter
       = BlacklistingTextInputFormatter(RegExp(r'\n'));
 }
 
+/// Old name for [FilteringTextInputFormatter.allow].
+// TODO(ianh): Deprecate these once the samples are migrated.
```
Comment (goderbauer) [inline packages/flutter/lib/src/services/text_formatter.dart:272]: Which samples is this referring to?

Comment (goderbauer) [reply packages/flutter/lib/src/services/text_formatter.dart:272]: Maybe add a link?

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by goderbauer** [comment]
Looks like cirrus is unhappy, though.

File: packages/flutter/lib/src/services/text_formatter.dart (hunk 2 above)
Comment (Hixie) [reply packages/flutter/lib/src/services/text_formatter.dart:272]: gallery in samples. they've said they'll fix it when i land this.

Comment (goderbauer) [reply packages/flutter/lib/src/services/text_formatter.dart:272]: https://github.com/flutter/gallery/blob/74d9d92883c6ca10742c44df309f065289cf2f55/lib/demos/material/text_field_demo.dart#L225

**Comment by christopherfujino** [comment]
Reverting this TBR by @Hixie as it caused LUCI post-submit to fail with:

https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8877105797085540816/+/steps/run_test.dart_for_framework_tests_shard/0/stdout

```
The following StateError was thrown running a test:
Bad state: No element

When the exception was thrown, this was the stack:
#0      Iterable.single (dart:core/iterable.dart:558:25)
#1      WidgetController._getElementPoint (package:flutter_test/src/controller.dart:646:47)
#2      WidgetController.getCenter (package:flutter_test/src/controller.dart:618:12)
#3      WidgetController.tap (package:flutter_test/src/controller.dart:256:18)
#4      main.<anonymous closure> (file:///b/s/w/ir/k/flutter/packages/flutter/test/material/text_field_test.dart:3285:18)
#15     FakeAsync.flushMicrotasks (package:fake_async/fake_async.dart:192:32)
#16     AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1135:17)
#17     AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1123:35)
(elided 25 frames from dart:async and package:stack_trace)

The test description was:
  Pasted values are formatted
```

**Comment by braulio94** [comment]
Since this is deprecated, what should we use instead?

**Comment by braulio94** [comment]
> Since this is deprecated, what should we use instead?

Got it, eg: 

**`FilteringTextInputFormatter.digitsOnly`**

**Pull request merged**